sources/main.cpp

```diff
@@ -132,6 +141,31 @@ static void execSimulation(boost::shared_ptr<dfl::Context> context, dfl::inputs:
   }
 }
 
+void dumpZipArchive(std::unordered_map<std::string, std::string> &mapOutputFilesData, boost::filesystem::path outputPath,
+                    const dfl::common::Options::RuntimeConfiguration &runtimeConfig) {
+  DYNAlgorithms::multiprocessing::Context &mpiContext = DYNAlgorithms::multiprocessing::context();
+  mpiContext.sync();
+  if (mpiContext.isRootProc()) {
+    bool outputIsZip = !runtimeConfig.zipArchivePath.empty();
+
+    if (outputIsZip) {
+      const std::string programLogFileRelativePath = runtimeConfig.programName + ".log";
+      const std::string programLogFileAbsolutePath = createAbsolutePath(programLogFileRelativePath, outputPath.generic_string());
+      dfl::common::Log::addLogFileContentInMapData(programLogFileRelativePath, programLogFileAbsolutePath, mapOutputFilesData);
+
+      boost::shared_ptr<zip::ZipFile> archive = zip::ZipFileFactory::newInstance();
+      const std::string archivePath = createAbsolutePath("outputs.zip", outputPath.generic_string());
+      if (!runtimeConfig.contingenciesFilePath.empty())
+        archive = zip::ZipInputStream::read(archivePath);
```

Comment:
Why is it needed to read the archive ?

Reply:
it is used when it was already created by dynawo-algorithms and we complete it